rubato/structure/gameobject/physics/engine.py

```diff
@@ -140,6 +140,9 @@ def collide(hitbox_a: Hitbox, hitbox_b: Hitbox) -> Optional[Manifold]:
         Returns:
             Returns a collision info object if a collision is detected or None if no collision is detected.
         """
+        if not hitbox_a.should_collide(hitbox_a, hitbox_b) or not hitbox_b.should_collide(hitbox_b, hitbox_a):
```

Comment:
```suggestion
        if not (hitbox_a.should_collide(hitbox_a, hitbox_b) and hitbox_b.should_collide(hitbox_b, hitbox_a)):
```
sorry previous hurts ma brain

Reply:
But it's less clear!!!